fix #51 - Parser should not fail on unknown elements

kaushikmupadhya opened the pull request:
Add changes to fix #51 to ensure the parser doesn't fail for unknown elements and add these elements to a list for future reference. Updating fix #50 into the other dependent files.

barredterra commented:
Thanks for the PR, @kaushikmupadhya.

Two questions:

- Why did you incorporate the proposed changes from #50 into this PR? Seems to me like these are independent changes which are better handled in separate PRs.
- What is supposed to happen with `self.unknown_elements`? Since there are not internal use, I guess it's meant for the library's user to analyze. In this case, should we add this info to the README?

kaushikmupadhya commented:
Hello Rafael,

Thank you for your questions!

1. I added the changes from #50 to this PR because they weren’t reflected
in my branch. I included them to test the proposed updates and see how they
work. I understand these could be handled in separate PRs, and I’m open to
making adjustments if needed.

2. For self.unknown_elements, I added them to allow users to store and
track unknown elements during XML serialization. This can be useful if, in
the future, there’s a need to automate handling or take care of these
elements. I think adding this information to the README would help users
understand its purpose and potential benefits.

I also want to mention the strict mode approach as discussed in #51. I
understand how important round-trip fidelity is, and I believe storing
unknown elements can highlight this feature.

Let me know if you need further changes or clarification!

Best regards,
Kaushik
______________________
On Mon, 9 Dec 2024, 15:35 Raffael Meyer, ***@***.***> wrote:

> Thanks for the PR, @kaushikmupadhya <https://github.com/kaushikmupadhya>.
>
> Two questions:
>
>    - Why did you incorporate the proposed changes from #50
>    <https://github.com/pretix/python-drafthorse/pull/50> into this PR?
>    Seems to me like these are independent changes which are better handled in
>    separate PRs.
>    - What is supposed to happen with self.unknown_elements? Since there
>    are not internal use, I guess it's meant for the library's user to analyze.
>    In this case, should we add this info to the README?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pretix/python-drafthorse/pull/52#issuecomment-2528130229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJCZX3PU47HEGBJXRMRM5ND2EWTB5AVCNFSM6AAAAABTHNDXR6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMRYGEZTAMRSHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

marination commented:
@kaushikmupadhya Some feedback:
1. The changes on https://github.com/pretix/python-drafthorse/pull/50 seem unrelated here. In fact, excluding those changes should create an environment where an error is thrown, and now this is perfect for you to implement your change where this error must be handled. This particular PR should aim to be a general fix.
2. "For self.unknown_elements, I added them to allow users to store and track unknown elements during XML serialization." If you have particular plans of making a new attribute available for usage, make the "why" and "how" of it clear in the PR description. It is a new feature for a library user.
3. Apart from `unknown_elements`, removing the raised validation is a breaking change. Ongoing implementations expect it to break, and your change would fail that expectation. If you have some design in mind for "strict_mode", elaborate on the plan of execution in discussion/review threads like these.
